challenge2.py

```diff
@@ -0,0 +1,82 @@
+"""
+    Advent of Code Day 2
+"""
+from itertools import groupby, combinations
+
+from common.input_file import read_strings
+
+
+def transform_input(box_id):
+    """
+        We sort and groupby to get lists of letters from each ID
+    """
+    grouped_box_id = groupby(sorted(box_id))
+
+    # We need to convert to a list so that we may go through this
+    # list more than once, otherwise we exhaust iterators first time through
+
+    return [(key, list(group)) for key, group in grouped_box_id]
+
+def get_checksum(box_ids):
+    """
+        Get the checksum of all the box IDs
+        Expects the box ids to be grouped and sorted
+    """
+    grouped_box_ids = [transform_input(box_id) for box_id in box_ids]
+
+    return (
+        get_number_of_boxes_with_exact_letters(grouped_box_ids, 2) *
+        get_number_of_boxes_with_exact_letters(grouped_box_ids, 3)
+        )
+
+def get_number_of_boxes_with_exact_letters(grouped_box_ids, desired_count):
+    """
+        Gets the count of boxes where we have exactly `desired_count`
+        number of one letter in it
+    """
+    candidates = [True for box in grouped_box_ids if contains_desired_count(box, desired_count)]
+    return len(candidates)
+
+def contains_desired_count(box, desired_count):
+    """
+        If the box has exatly 'desired_count` of any one letter
```

Comment:
one issue with comments is that it allows inconsistent spelling mistakes ;)

Reply:
yup, thanks